src/main/java/com/thealgorithms/bitmanipulation/HighestSetBit.java

```diff
@@ -0,0 +1,26 @@
+package com.thealgorithms.bitmanipulation;
+
+/**
+ * Find Highest Set Bit
+ * @author Bama Charan Chhandogi (https://github.com/BamaCharanChhandogi)
+ */
+
+public class HighestSetBit {
+    public static int findHighestSetBit(int num) {
+        if (num < 0) {
+            throw new IllegalArgumentException("Input cannot be negative");
+        }
+
+        if (num == 0) {
+            return -1; // No set bits
+        }
```

Comment:
What do you think about using [`Optional`](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) here? Please have a look at the _main_ comment.

Reply:
@BamaCharanChhandogi feel free to mark is as _resolved_.